### ISSUE TITLE
Create patterns + plus icon button: improve labeling and implementation

### ISSUE THREAD

**Issue by afercia** [WordPress/gutenberg#52637]
### Description

In the Site Editor > Patterns > the + plus icon button is labelled 'Create pattern'. Its tooltip uses the same text:

<img width="382" alt="Screenshot 2023-07-13 at 14 44 20" src="https://github.com/WordPress/gutenberg/assets/1682452/ee543b1a-6682-4317-b1a0-8df3b57e7851">

However, clicking the button reveals a menu with _two_ controls:
- Create template part 
- Create pattern

<img width="498" alt="Screenshot 2023-07-13 at 14 44 24" src="https://github.com/WordPress/gutenberg/assets/1682452/d459ba36-5a02-43af-ac7a-89508055a8d7">

As such, the button label / tooltip 'Create pattern' is incorrect or, at the very least, incomplete.

I'm not sure why this icon button uses the + plus icon in the first place. That's inconsistent with other + plus buttons, for example the 'Draft a new page' one: it gives immediate access to the modal dialog where users can draft a new page. 

<img width="395" alt="Screenshot 2023-07-14 at 14 16 56" src="https://github.com/WordPress/gutenberg/assets/1682452/2e31254b-6497-4827-876b-8ae4de91ffb9">

To recap:
The two buttons look the same. As a user, I would expect the same interaction and user experience. Instead:
- 'Create pattern' opens a dropdown menu.
- 'Draft a new page' opens a modal dialog.

I'm not sure this is the best UI. To me, controls that look the same and hint to the same action should also behave the same.

I think a more appropriate icon for the 'Create pattern'  button would be the vertical ellipsis button. That's usually associated with the concept of 'more actions' so that users expect to see _more than one option_ when activating the ellipsis button.

Alternatively, the two available actions 'Create template part' and 'Create pattern' should be presented separately.

### Step-by-step reproduction instructions

- Go to the Site Editor > Patterns > Create pattern + plus icon button
- Click the button.
- Observe that a menu opens with two options.
- Observe that 'Create template part' is unexpected there, as the button you clicked to open the menu only mentions patterns.

### Please confirm that you have searched existing issues in the repo.

Yes

### Please confirm that you have tested with all plugins deactivated except Gutenberg.

Yes
